invenio_rdm_records/alembic/2186256e8d9b_add_deletion_status_to_rdmrecords.py

```diff
@@ -0,0 +1,76 @@
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Add deletion status to RDMRecords."""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "2186256e8d9b"
+down_revision = "ff860d48fb4b"
+branch_labels = ()
+depends_on = None
+
+
+def _get_default_deletion_status():
+    """Try to get the default value for record deletion status."""
+    try:
+        # first try: get the default value from the DB model class
+        from invenio_rdm_records.records.models import RDMRecordMetadata
+
+        default_value = RDMRecordMetadata.deletion_status.default.arg
+        assert len(default_value) == 1
+
+    except Exception:
+        try:
+            # second try: try to get the value from the enum (more flaky)
+            from invenio_rdm_records.records.systemfields.deletion_status import (
+                RecordDeletionStatusEnum,
+            )
+
+            default_value = RecordDeletionStatusEnum.PUBLISHED.value
+            assert len(default_value) == 1
+
+        except Exception:
+            # fallback: just use 'P', as that's the current default (most flaky)
+            default_value = "P"
+
+    return default_value
+
+
+def upgrade():
+    """Upgrade database."""
+    # step 1: create the columns, but make them nullable for now
+    op.add_column(
+        "rdm_records_metadata",
+        sa.Column("deletion_status", sa.String(length=1), nullable=True),
+    )
+    op.add_column(
+        "rdm_records_metadata_version",
+        sa.Column(
+            "deletion_status",
+            sa.String(length=1),
+            nullable=True,
+        ),
+    )
+
+    # step 2: set default values for existing rows
+    default_value = _get_default_deletion_status()
+    op.execute(f"UPDATE rdm_records_metadata SET deletion_status = '{default_value}'")
+    op.execute(
+        f"UPDATE rdm_records_metadata_version SET deletion_status = '{default_value}'"
+    )
```

Comment:
To avoid manually building SQL strings (not that we'll ever get SQL injection attacks in this part), you can check out [this example](https://github.com/inveniosoftware/invenio-sipstore/blob/c213d0738360f44eebd643f2e095a2ce53bd9dcc/invenio_sipstore/alembic/b31cad2f14c7_extend_sip_metadata_model.py#L79-L81).